Add ``estimate_qpu_access_time`` method

JoelPasvolsky opened the pull request:
I still need to unit tests next week but would like @gmarsden to take a look at the formulas.

gmarsden commented on dwave/system/samplers/dwave_sampler.py:
```diff
@@ -522,3 +524,221 @@ def to_networkx_graph(self):
+            decorrelation_max_nominal_anneal_time = problem_timing_data['decorrelation_max_nominal_anneal_time']
+            decorrelation_time_range = problem_timing_data['decorrelation_time_range']
+
+            default_readout_thermalization = self.properties['default_readout_thermalization']
```
This seems odd. "default_readout_thermalization" is in the problem_timing_data too, isn't it?

JoelPasvolsky replied:
it is, and I can take it from there

JoelPasvolsky replied:
Done. Also added missing return value in docstring.

randomir reviewed (commented):
Haven't reviewed it for real yet, but some random thoughts:
- accepting same parameters as `.sample()` methods would be nice (inferring number of qubits from problem submitted, etc)
- adding this estimator to the cloud client would enable easier reuse by e.g. AWS Braket
- we should use this to add some realistic timing info to our mock QPU samplers

JoelPasvolsky commented:
> * accepting same parameters as `.sample()` methods would be nice (inferring number of qubits from problem submitted, etc)

Do you mean providing the estimate when someone submits a problem (we want to estimate without submitting) or learning the number of qubits from a BQM (will vary because is heuristic) or something else?

Let's talk about cloud-client versus DWaveSampler

gmarsden commented:
> Do you mean providing the estimate when someone submits a problem (we want to estimate without submitting) or learning the number of qubits from a BQM (will vary because is heuristic) or something else?

I read this as having `estimate_qpu_access_time` take the same parameters as `sample` so that the user doesn't have to do work to translate to the parameters (doesn't need to count qubits, set flags, etc) -- just passes the same params.

JoelPasvolsky commented:
@randomir, I addressed your comment's 1st bullet by replacing Boolean ``reverse_anneal`` with a check on actual param ``anneal_schedule`` for reverse annealing, so now the method can accept the same params as the ``sample()`` method.
I did not yet default to having the method estimate the number of embedded qubits because that can be slow and if we move this to cloud-client we'd need to add minorminer there too and also because whether or not we should do this depends on the user's workflow

randomir commented:
Thanks, @JoelPasvolsky. Having a low-level estimator like this -- only encoding the (versioned) rule, without any smarts like embedding -- is a great start. I wasn't trying to suggest we should embed a logical problem here just for time estimation's sake. That would an absolute overkill (given embedding's complexity).

However, I still think this function (in its current form) is better suited in the cloud-client (for reusability), but I could be convinced otherwise.

JoelPasvolsky commented:
Ported to [dwave-cloud-client](https://github.com/dwavesystems/dwave-cloud-client/pull/530)
